pool/app/contact/repo/repo_sql.ml

```diff
@@ -340,14 +343,64 @@ let find_filtered pool ?order_by ?limit experiment_id filter =
   let (Dynparam.Pack (pt, pv)) = dyn in
   let open Caqti_request.Infix in
   let request =
-    sql |> find_filtered_request_sql ?limit |> pt ->* Repo_model.t
+    sql
+    |> where_prefix
+    |> find_filtered_request_sql ?limit
+    |> pt ->* Repo_model.t
   in
   let%lwt contacts =
     Utils.Database.collect (pool |> Pool_database.Label.value) request pv
   in
   Lwt_result.return contacts
 ;;
 
+let matches_filter ?(default = false) pool experiment_id filter contact =
```

Comment:
Do you think it is possible to include this filter in the `find_all_public_by_contact_request` function? In case we add pagination, it would still work correctly.